### DISCUSSION TITLE
compiling always stuck at 95%

### DISCUSSION THREAD

**Discussion by dirkChalet** [transmission/transmission#4360] [Support]
[ 95%] Building CXX object tests/libtransmission/CMakeFiles/libtransmission-test.dir/web-utils-test.cc.o

been like that for over 12hours.
Linux GS75 5.19.0-26-generic #27-Ubuntu SMP PREEMPT_DYNAMIC Wed Nov 23 20:44:15 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux

I reinstalled QT. think I am using qt5 now. Was also seeing strange crashes. like clicking and trying to stop a torrent gives sometimes instant crash. I think its a UI problem. It's how I came looking for 4.0, as in the 3 version the properties UI is all wrong, with wrong text in the boxes etc... 

I tried several recent versions for compiling. always stuck at same file. compiling also seems slower than before.
Any help moving me forward would be appreciated. 
Dirk
edit : I forgot to say that I had no problem compiling until a few days ago.

**Comment by dirkChalet** [comment]
Left again overnight with new build and stuck again at same point. so manual work around like this : ( I pressed control-c to stop it )
make transmission-qt
make transmission-create
make transmission-daemon
make transmission-edit
make transmission-remote
make transmission-show

sudo cp ./utils/transmission-* /usr/local/bin
sudo cp ./daemon/transmission-daemon /usr/local/bin/
sudo cp ./qt/transmission-qt /usr/local/bin/

I hope no library needed to be copied :-) . but the qt version runs for me.
Maybe this helps someone.
Dirk

**Comment by rsekman** [comment]
I am having the same issue. By checking with `top` I found that the build gets stuck running `clang-tidy`. I think this may be due to an upstream change in `clang-tidy` because
* the issue appeared for me after upgrading from `14.0.6` to `15.0.7`,
* `web-utils-test.cc` has not been changed since my last successful build, and
*  I can't now build commits that I could build before. 

I am on Arch and I am not sure how to check version histories for Ubuntu -- did you also recently upgrade `clang-tidy`? You should be able to check by grepping through `/var/log/apt`.

In the meantime, you can work around this issue by passing `-DRUN_CLANG_TIDY=OFF` to `cmake` to skip running `clang-tidy`.

**Comment by dirkChalet** [comment]
I haven’t compiled since.Thank you for the work around. I will try it later.DirkSent from my iPadOn 11 Feb 2023, at 09:57, Robin Seth Ekman ***@***.***> wrote:﻿
I am having the same issue. By checking with top I found that the build gets stuck running clang-tidy. I think this may be due to an upstream change in clang-tidy because

the issue appeared for me after upgrading from 14.0.6 to 15.0.7,
web-utils-test.cc has not been changed since my last successful build, and
I can't now build commits that I could build before.

I am on Arch and I am not sure how to check version histories for Ubuntu -- did you also recently upgrade clang-tidy? You should be able to check by grepping through /var/log/apt.
In the meantime, you can work around this issue by passing -DRUN_CLANG_TIDY=OFF to cmake to skip running clang-tidy.

**Comment by ckerr** [comment]
This does seem to be a clang-tidy bug. I'm seeing the same behavior when switching between clang-tidy-14 (works) and clang-tidy-15 (hangs).

There's an issue tracking this at https://github.com/transmission/transmission/issues/4821 and a PR to fix it in https://github.com/transmission/transmission/pull/4824 for 4.0.1.

@dirkChalet is correct that `-DRUN_CLANG_TIDY=OFF` is the workaround. :bowing_man:

**Comment by ckerr** [comment]
Locking as a duplicate of #4821
